import_ldraw.py

```diff
@@ -928,6 +946,41 @@ def getColorValue(line, value):
     if value in line:
         n = line.index(value)
         return line[n + 1]
+        
+# Clean-up the design by linking identical parts (mesh/colour)        
+def linkedParts():
+    colours = []
+    for colour in bpy.data.materials:
+        colours.append(colour.name)
+    
+    parts = []
+    for part in bpy.data.objects:
+        if part.type == 'MESH' and part.name.split('.')[0] not in parts:
+            parts.append(part.name.split('.')[0])
+
+    for part in parts:
+        for colour in colours:
+            replaceParts(part, colour)
+
+# Replace identical meshes of part/colour-combination with a linked version   
+def replaceParts(Part, Colour):
+    scene = bpy.context.scene
+    mat = bpy.data.materials[Colour]
+    mesh = ''
+
+    for ob in scene.objects:
+        if ob.type == 'MESH' and ob.name.split('.')[0] == Part:
+            for slot in ob.material_slots:
+                if slot.material == mat:
+                    if mesh == '':
+                        mesh = ob.data
+                    else:
+                        ob.data = mesh    
+                    ob.select = True
+        else: 
+            ob.select = False
+    if mesh != '':
+        mesh.name = Part + ' ' + Colour           
```

Comment:
This is overloading the addition operation and should never bee done. Instead, use `mesh.name = "{0} {1}".format(Part, Color)`. :)